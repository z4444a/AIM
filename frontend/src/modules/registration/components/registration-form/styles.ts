import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';
import { WithStyles } from '@material-ui/core/styles/withStyles';

export const styles = (theme: Theme) =>
  createStyles({
    container: {
      display: 'flex',
      width: '100vw',
      justifyContent: 'center',
      alignItems: 'center',
    },
    paper: {
      marginTop: theme.spacing.unit * 8,
    },
    card: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      width: '500px',
      padding: theme.spacing.unit * 3,
    },
    avatar: {
      margin: theme.spacing.unit,
      backgroundColor: theme.palette.secondary.main,
    },
    form: {
      width: '100%',
      marginTop: theme.spacing.unit,
    },
    greeting: {
      marginBottom: theme.spacing.unit * 2,
    },
    submit: {
      marginTop: theme.spacing.unit * 3,
    },
  });
export type Style = WithStyles<
  'container' | 'paper' | 'card' | 'avatar' | 'form' | 'submit' | 'greeting'
>;
