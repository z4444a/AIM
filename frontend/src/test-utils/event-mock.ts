export const getMockChangeEvent = (value: string) => {
  return {
    target: {
      value: value,
    },
  };
};

export const getMockKeyPressEvent = (key: string) => {
  return {
    key: key,
    preventDefault: jest.fn().mockName('preventDefault'),
  };
};
