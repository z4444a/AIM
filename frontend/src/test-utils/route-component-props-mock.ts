import { RouteComponentProps } from 'react-router';
import { Href } from 'history';

export function getMockRouterProps<P>(data: P): RouteComponentProps {
  const location = {
    hash: '',
    key: '',
    pathname: '',
    search: '',
    state: {},
  };

  const props: RouteComponentProps<P> = {
    match: {
      isExact: true,
      params: data,
      path: '',
      url: '',
    },
    location: location,
    history: {
      length: 2,
      action: 'POP',
      location: location,
      push: () => {},
      replace: () => {},
      go: () => {},
      goBack: () => {},
      goForward: () => {},
      block: () => () => {},
      createHref: () => {
        const temp: Href = '';
        return temp;
      },
      listen: () => () => {},
    },
    staticContext: {},
  };

  return props;
}
