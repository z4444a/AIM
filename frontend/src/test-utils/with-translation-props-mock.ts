import { WithTranslation } from 'react-i18next';

export function getMockWithTranslationProps(): WithTranslation {
  // eslint-disable-next-line
  const t = ((key: string) => `${key}_i18n`) as any;
  // eslint-disable-next-line
  const i18n = {} as any;
  const props: WithTranslation = {
    tReady: true,
    t,
    i18n,
  };

  return props;
}
