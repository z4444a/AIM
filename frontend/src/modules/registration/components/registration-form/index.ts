import withStyles from '@material-ui/core/styles/withStyles';
import { styles } from './styles';
import { RegistrationForm } from './registration-form';
import { withTranslation } from 'react-i18next';

export default withStyles(styles)(withTranslation('common')(RegistrationForm));
