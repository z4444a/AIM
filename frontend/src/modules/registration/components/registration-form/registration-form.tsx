import React, { KeyboardEvent } from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CssBaseline from '@material-ui/core/CssBaseline';
import IconButton from '@material-ui/core/IconButton';
import InputAdornment from '@material-ui/core/InputAdornment';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import SupervisorAccount from '@material-ui/icons/SupervisorAccount';
import Visibility from '@material-ui/icons/Visibility';
import VisibilityOff from '@material-ui/icons/VisibilityOff';
import { Style } from './styles';
import { WithTranslation } from 'react-i18next';

export interface FormData {
  login: string;
  pwd: string;
  confirmPwd: string;
}

type InputType = keyof FormData;

export interface Props {
  value: FormData;
  onValueChanged?: (newValue: FormData) => void;
  submitRegistration?: (newValue: FormData) => void;
  showPwd: boolean;
  toggleShowPwd?: (showPwd: boolean) => void;
  showPwdConfirm: boolean;
  toggleShowPwdConfirm?: (showPwd: boolean) => void;
}

interface State {
  isPwdConfirmFieldDirty: boolean;
}

export type InternalProps = Style & Props & WithTranslation;

export class RegistrationForm extends React.PureComponent<InternalProps, State> {
  public constructor(props: InternalProps) {
    super(props);
    this.state = {
      isPwdConfirmFieldDirty: false,
    };
  }

  public render(): JSX.Element {
    const { classes, value, showPwd, showPwdConfirm, t } = this.props;
    const { isPwdConfirmFieldDirty } = this.state;
    return (
      <div className={classes.container}>
        <CssBaseline />
        <div className={classes.paper}>
          <Card className={classes.card}>
            <Avatar className={classes.avatar}>
              <SupervisorAccount />
            </Avatar>
            <Typography classes={{ root: classes.greeting }} component="h1" variant="h6">
              {t('registrationPage.form.header')}
            </Typography>
            <form className={classes.form} noValidate>
              <TextField
                id="login"
                label={t('registrationPage.form.loginFieldLabel')}
                fullWidth
                autoFocus
                required
                margin="dense"
                variant="outlined"
                value={value.login}
                onChange={this.handleLoginChange}
                onKeyPress={this.handleLoginKeyPress}
              />
              <TextField
                id="password"
                value={value.pwd}
                fullWidth
                required
                label={t('registrationPage.form.pwdFieldLabel')}
                margin="dense"
                variant="outlined"
                onChange={this.handlePwdChange}
                onKeyPress={this.handlePwdKeyPress}
                type={showPwd ? 'text' : 'password'}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <IconButton
                        tabIndex={-1}
                        aria-label="Toggle password visibility"
                        onClick={this.toggleShowPwd}
                      >
                        {showPwd ? <VisibilityOff /> : <Visibility />}
                      </IconButton>
                    </InputAdornment>
                  ),
                }}
              />
              <TextField
                id="passwordConfirm"
                value={value.confirmPwd}
                fullWidth
                required
                label={t('registrationPage.form.pwdConfirmFieldLabel')}
                margin="dense"
                variant="outlined"
                onChange={this.handlePwdConfirmChange}
                onKeyPress={this.handlePwdConfirmKeyPress}
                type={showPwdConfirm ? 'text' : 'password'}
                error={value.pwd !== value.confirmPwd && isPwdConfirmFieldDirty}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <IconButton
                        tabIndex={-1}
                        aria-label="Toggle password visibility"
                        onClick={this.toggleShowPwdConfirm}
                      >
                        {showPwdConfirm ? <VisibilityOff /> : <Visibility />}
                      </IconButton>
                    </InputAdornment>
                  ),
                }}
              />
              <Button
                className={classes.submit}
                fullWidth
                variant="contained"
                onClick={this.submitRegistration}
                disabled={!value.login || !value.pwd || value.pwd !== value.confirmPwd}
              >
                {t('registrationPage.form.submitBtn')}
              </Button>
            </form>
          </Card>
        </div>
      </div>
    );
  }

  private handleValueChanged(inputType: InputType, event: React.ChangeEvent<HTMLInputElement>) {
    const { value, onValueChanged } = this.props;
    if (typeof onValueChanged !== 'function') {
      return;
    }
    const newValue: FormData = Object.assign({}, value);
    newValue[inputType] = event.target.value;

    onValueChanged(newValue);
  }

  private handleKeyPress(inputType: InputType, event: KeyboardEvent) {
    if (event.key === 'Enter') {
      this.submitRegistration();
      event.preventDefault();
    }
  }

  private submitRegistration = () => {
    const { submitRegistration, value } = this.props;
    if (typeof submitRegistration !== 'function') {
      return;
    }
    submitRegistration(value);
  };

  private handleLoginChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    this.handleValueChanged('login', event);
  private handlePwdChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    this.handleValueChanged('pwd', event);
  private handlePwdConfirmChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    this.handleValueChanged('confirmPwd', event);
  private handleLoginKeyPress = (event: KeyboardEvent) => this.handleKeyPress('login', event);
  private handlePwdKeyPress = (event: KeyboardEvent) => this.handleKeyPress('pwd', event);
  private handlePwdConfirmKeyPress = (event: KeyboardEvent) => {
    this.setState({ isPwdConfirmFieldDirty: true });
    this.handleKeyPress('confirmPwd', event);
  };
  private toggleShowPwd = () => {
    const { showPwd, toggleShowPwd } = this.props;
    if (typeof toggleShowPwd !== 'function') {
      return;
    }
    toggleShowPwd(!showPwd);
  };
  private toggleShowPwdConfirm = () => {
    const { showPwdConfirm, toggleShowPwdConfirm } = this.props;
    if (typeof toggleShowPwdConfirm !== 'function') {
      return;
    }
    toggleShowPwdConfirm(!showPwdConfirm);
  };
}
