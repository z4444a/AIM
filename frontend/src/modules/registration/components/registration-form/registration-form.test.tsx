import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { FormData, InternalProps, RegistrationForm } from './registration-form';
import { getMockWithTranslationProps } from '../../../../test-utils/with-translation-props-mock';

describe('RegistrationForm', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const data: FormData = {
      login: '',
      pwd: '',
      confirmPwd: '',
    };
    const classes = {
      container: 'container',
      paper: 'paper',
      avatar: 'avatar',
      form: 'form',
      submit: 'submit',
      card: 'card',
      greeting: 'greeting',
    };
    const withTranslationProps = getMockWithTranslationProps();

    const props: InternalProps = {
      value: data,
      showPwd: false,
      showPwdConfirm: false,
      classes,
      ...withTranslationProps,
    };
    wrapper = shallow(<RegistrationForm {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
