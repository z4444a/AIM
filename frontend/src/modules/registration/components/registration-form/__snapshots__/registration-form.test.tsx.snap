// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm should render correctly 1`] = `
<div
  className="container"
>
  <WithStyles(CssBaseline) />
  <div
    className="paper"
  >
    <WithStyles(Card)
      className="card"
    >
      <WithStyles(Avatar)
        className="avatar"
      >
        <pure(SupervisorAccountIcon) />
      </WithStyles(Avatar)>
      <WithStyles(Typography)
        classes={
          Object {
            "root": "greeting",
          }
        }
        component="h1"
        variant="h6"
      >
        registrationPage.form.header_i18n
      </WithStyles(Typography)>
      <form
        className="form"
        noValidate={true}
      >
        <TextField
          autoFocus={true}
          fullWidth={true}
          id="login"
          label="registrationPage.form.loginFieldLabel_i18n"
          margin="dense"
          onChange={[Function]}
          onKeyPress={[Function]}
          required={true}
          select={false}
          value=""
          variant="outlined"
        />
        <TextField
          InputProps={
            Object {
              "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                position="end"
              >
                <WithStyles(IconButton)
                  aria-label="Toggle password visibility"
                  onClick={[Function]}
                  tabIndex={-1}
                >
                  <pure(VisibilityIcon) />
                </WithStyles(IconButton)>
              </WithStyles(WithFormControlContext(InputAdornment))>,
            }
          }
          fullWidth={true}
          id="password"
          label="registrationPage.form.pwdFieldLabel_i18n"
          margin="dense"
          onChange={[Function]}
          onKeyPress={[Function]}
          required={true}
          select={false}
          type="password"
          value=""
          variant="outlined"
        />
        <TextField
          InputProps={
            Object {
              "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
                position="end"
              >
                <WithStyles(IconButton)
                  aria-label="Toggle password visibility"
                  onClick={[Function]}
                  tabIndex={-1}
                >
                  <pure(VisibilityIcon) />
                </WithStyles(IconButton)>
              </WithStyles(WithFormControlContext(InputAdornment))>,
            }
          }
          error={false}
          fullWidth={true}
          id="passwordConfirm"
          label="registrationPage.form.pwdConfirmFieldLabel_i18n"
          margin="dense"
          onChange={[Function]}
          onKeyPress={[Function]}
          required={true}
          select={false}
          type="password"
          value=""
          variant="outlined"
        />
        <WithStyles(Button)
          className="submit"
          disabled={true}
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          registrationPage.form.submitBtn_i18n
        </WithStyles(Button)>
      </form>
    </WithStyles(Card)>
  </div>
</div>
`;
